app/wama.web/Controllers/CheckInController.cs

```diff
@@ -44,6 +47,14 @@ public IActionResult Index()
                 {
                     SetErrorMessages(AppString.AccountPendingApprovalMessage);
                 }
+                else if (userAccount.IsSuspended)
+                {
+                    SetErrorMessages(AppString.AccountSuspended);
+                }
+                else if (certificateInfo != null && System.DateTimeOffset.Compare(System.DateTimeOffset.Now, certificateInfo.ExpiresOn) > 0)
+                {
+                    SetErrorMessages(AppString.CertificateExpired);
```

Comment:
When a certification expires, a patron can still login.
Hence, you should figure out a means to pass a warning message to the Successful page.